.github/workflows/npm-publish.yml

```diff
@@ -4,191 +4,238 @@ on:
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Setup Node
         uses: actions/setup-node@v1
         with:
           node-version: 16
-      - name: Checkout `main` branch
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Checkout main branch
         uses: actions/checkout@v2
         with:
-          token: ${{ secrets.GH_TOKEN }}
           ref: main
-      - name: Configure git
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Configure git
         run: |
           git config user.email "${{ github.actor }}@users.noreply.github.com"
           git config user.name ${{ github.actor }}
           git fetch --all
-      - name: Merge down from `main` -> `rc` -> `develop`
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Merge down from main -> rc -> develop
         run: |
           git checkout -B release-candidate refs/remotes/origin/release-candidate
           git merge main
           git checkout -B develop refs/remotes/origin/develop
           git merge release-candidate
           git checkout main
-      - name: Merge `release-candidate` -> `main`
-        if: ${{ steps.set-env.outputs.main == 'true' && steps.set-env.outputs.hotfix != 'true' }}
+      -
+        name: Merge rc -> main
         run: git merge release-candidate
-      - name: Initialize and build code
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Install Dagger
+        uses: dagger/dagger-for-github@v3
+        with:
+          install-only: true
+      -
+        name: Verify Docker image
+        run: |
+          dagger project init
+          dagger project update
+          dagger project update "github.com/3box/pipeline-tools/ci@v${{ env.PIPELINE_TOOLS_VER }}"
+          dagger do verify -p ${{ env.DAGGER_PLAN }}
+      -
+        name: Initialize and build code
         run: |
           npm set unsafe-perm true
           npm ci && npm run build
-          git commit -m 'chore(core): bump version' -- packages/core/src/version.ts || true
-      - name: Publish packages to NPM
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Publish packages to NPM
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-          GH_TOKEN: ${{ secrets.GH_TOKEN }}
-          PUBLISH_ARGS: ${{ steps.set-env.outputs.publishArgs }}
+          GH_TOKEN: ${{ env.GH_TOKEN }}
         run: |
           npm config set //registry.npmjs.org/:_authToken $NODE_AUTH_TOKEN
-          npm run publish:latest $PUBLISH_ARGS
-          git commit -m 'chore(core): bump version' -- packages/core/src/version.ts || true
-          git push
-      - name: Update and push code to `release-candidate` and `develop`
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+          npx lerna publish --allow-branch main --dist-tag latest --no-verify-access --conventional-graduate bump minor --yes
```

Comment:
is there any way to re-use the `publish:release` command from the package.json so we don't need to keep the command in sync in 2 different places?

Reply:
Yeah, I was thinking about that. The `npm` command will run from whichever branch is checked-out, so it might not be one that has the very latest `package.json` changes to any of the `publish` commands.

However, this is likely just temporary till all the branches have the latest file, and these commands are unlikely to change often.

For next week, we might have to make another RC (even if not necessary otherwise) so that these new commands are merged into the RC branch, _then_ make the release, _then_ make the new RC.

Reply:
Could we separate the changes to the package.json from the github workflow changes into two PRs, and merge/release the first before the second?

Reply:
We could but we usually promote changes to rc/main through the release process. To get just the package.json changes up, we'd have to do the same thing - make the RC to get it to rc branch, then release to get to main, then another RC.

Or maybe I'm missing what you're thinking about how this could work?

Reply:
Actually, I understand what you mean. Is it worth the trouble though? We'd have to make RCs and releases twice to test the changes anyway, even if we merge just the package changes in the first round.